build-logic/root-build/src/main/kotlin/gradlebuild.warmup-ec2.gradle.kts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+tasks.register("resolveAllDependencies") {
```

Comment:
This task does not resolve all the dependencies of the included builds. Is this a problem? Would it be better to have a task `resolveAllDependencies` in each sub-project, and then you could still run the task without using the `allprojects` block?

Reply:
Why we should avoid using `allprojects`?

Reply:
In general, you shouldn't access things from other projects. And that is what we do here - we access all configurations of all other projects.